{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "sns.set_context('notebook')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average* per year of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Country</th>\n",
       "      <th>Daily average of deaths</th>\n",
       "      <th>Daily average of new cases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Guinea</td>\n",
       "      <td>4.142857</td>\n",
       "      <td>21.681818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Liberia</td>\n",
       "      <td>23.159090</td>\n",
       "      <td>207.546387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Sierra-Leone</td>\n",
       "      <td>773.343750</td>\n",
       "      <td>46.084209</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Country  Daily average of deaths  Daily average of new cases\n",
       "0        Guinea                 4.142857                   21.681818\n",
       "1       Liberia                23.159090                  207.546387\n",
       "2  Sierra-Leone               773.343750                   46.084209"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Change to average per month !\n",
    "\n",
    "#Guinea\n",
    "#list the c files for Guinea\n",
    "filelist_Guinea = os.listdir('Data/ebola/guinea_data/');\n",
    "#create a list of tables containing the data from the csv files, choose description as index\n",
    "list_of_tables_Guinea = [pd.read_csv('Data/ebola/guinea_data/' + file,index_col = 'Description') for file in filelist_Guinea];\n",
    "#concatenate list\n",
    "Guinea = pd.concat(list_of_tables_Guinea)\n",
    "#select Totals which gives table of indeces(Description) and totals\n",
    "Guinea = Guinea['Totals']\n",
    "#choose The desired descriptions\n",
    "Guinea_new_cases = Guinea.loc['Total new cases registered so far'].astype(np.float32)\n",
    "Guinea_deaths = Guinea.loc['New deaths registered'].astype(np.float32)\n",
    "\n",
    "#Liberia\n",
    "filelist_Liberia = os.listdir('Data/ebola/liberia_data/');\n",
    "list_of_tables_Liberia = [pd.read_csv('Data/ebola/liberia_data/' + file,index_col = 'Variable') for file in filelist_Liberia];\n",
    "Liberia = pd.concat(list_of_tables_Liberia)\n",
    "Liberia = Liberia['National']\n",
    "#The value returned here is too large. I didn't take the time to look into it.\n",
    "Liberia_new_cases = Liberia.loc['New Case/s (Suspected)'].astype(np.float32)\n",
    "Liberia_deaths = Liberia.loc['Newly reported deaths'].astype(np.float32)\n",
    "\n",
    "#Sierra Leone\n",
    "filelist_SL = os.listdir('Data/ebola/sl_data/');\n",
    "list_of_tables_SL = [pd.read_csv('Data/ebola/sl_data/' + file,index_col = 'variable') for file in filelist_SL];\n",
    "SL = pd.concat(list_of_tables_SL)\n",
    "SL = SL['National']\n",
    "SL_new_cases = SL.loc['new_confirmed'].astype(np.float32)\n",
    "#For the deaths of Sierra Leone I know that I chose the wrong one.. We need to discuss what to do in this case\n",
    "SL_deaths = SL.loc['death_confirmed'].astype(np.float32)\n",
    "\n",
    "Ebola = pd.DataFrame({'Country':['Guinea', 'Liberia', 'Sierra-Leone'],\n",
    "                     'Daily average of new cases':[Guinea_new_cases.mean(),Liberia_new_cases.mean(),SL_new_cases.mean()],'Daily average of deaths':[Guinea_deaths.mean(),Liberia_deaths.mean(),SL_deaths.mean()]})\n",
    "Ebola"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Count</th>\n",
       "      <th>Group</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BARCODE</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MID1_0</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>7</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_1</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_2</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...</td>\n",
       "      <td>3</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_3</th>\n",
       "      <td>Archaea \"Crenarchaeota\" Thermoprotei Thermopro...</td>\n",
       "      <td>3</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_4</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>7</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_5</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_6</th>\n",
       "      <td>Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_7</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_8</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_9</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_10</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_11</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_12</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_13</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Halobacteria Halobacte...</td>\n",
       "      <td>4</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_14</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanococci Methanoco...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_15</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Methanopyri Methanopyr...</td>\n",
       "      <td>12</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_16</th>\n",
       "      <td>Archaea \"Euryarchaeota\" Thermoplasmata Thermop...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_17</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_18</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_19</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_20</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_21</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_22</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>53</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_23</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_24</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_25</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>3</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_26</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>1</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_27</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>99</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_28</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>17</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID1_29</th>\n",
       "      <td>Bacteria \"Actinobacteria\" Actinobacteria Actin...</td>\n",
       "      <td>2</td>\n",
       "      <td>EXTRACTION CONTROL</td>\n",
       "      <td>unknown</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_251</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>10</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_252</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_253</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_254</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_255</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>12</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_256</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>4</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_257</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>805</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_258</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>862</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_259</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>12</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_260</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3534</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_261</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>4</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_262</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_263</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>25</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_264</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_265</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_266</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_267</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_268</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>3</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_269</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>2</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_270</th>\n",
       "      <td>Bacteria \"Proteobacteria\" Gammaproteobacteria ...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_271</th>\n",
       "      <td>Bacteria \"Synergistetes\" Synergistia Synergist...</td>\n",
       "      <td>3</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_272</th>\n",
       "      <td>Bacteria \"Tenericutes\" Mollicutes Anaeroplasma...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_273</th>\n",
       "      <td>Bacteria \"Tenericutes\" Mollicutes Haloplasmata...</td>\n",
       "      <td>11</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_274</th>\n",
       "      <td>Bacteria \"Thermodesulfobacteria\" Thermodesulfo...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_275</th>\n",
       "      <td>Bacteria \"Thermodesulfobacteria\" Thermodesulfo...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_276</th>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>43</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_277</th>\n",
       "      <td>Bacteria \"Thermotogae\" Thermotogae Thermotogal...</td>\n",
       "      <td>16</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_278</th>\n",
       "      <td>Bacteria \"Verrucomicrobia\" Opitutae Opitutales...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_279</th>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>1</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MID9_280</th>\n",
       "      <td>Bacteria Cyanobacteria Cyanobacteria  Chloropl...</td>\n",
       "      <td>10</td>\n",
       "      <td>Control 2</td>\n",
       "      <td>stool</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2396 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                       Name  Count  \\\n",
       "BARCODE                                                              \n",
       "MID1_0    Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      7   \n",
       "MID1_1    Archaea \"Crenarchaeota\" Thermoprotei Desulfuro...      2   \n",
       "MID1_2    Archaea \"Crenarchaeota\" Thermoprotei Sulfoloba...      3   \n",
       "MID1_3    Archaea \"Crenarchaeota\" Thermoprotei Thermopro...      3   \n",
       "MID1_4    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      7   \n",
       "MID1_5    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      1   \n",
       "MID1_6    Archaea \"Euryarchaeota\" \"Methanomicrobia\" Meth...      1   \n",
       "MID1_7    Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...      1   \n",
       "MID1_8    Archaea \"Euryarchaeota\" Archaeoglobi Archaeogl...      1   \n",
       "MID1_9    Archaea \"Euryarchaeota\" Halobacteria Halobacte...      1   \n",
       "MID1_10   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      4   \n",
       "MID1_11   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      1   \n",
       "MID1_12   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      2   \n",
       "MID1_13   Archaea \"Euryarchaeota\" Halobacteria Halobacte...      4   \n",
       "MID1_14   Archaea \"Euryarchaeota\" Methanococci Methanoco...      1   \n",
       "MID1_15   Archaea \"Euryarchaeota\" Methanopyri Methanopyr...     12   \n",
       "MID1_16   Archaea \"Euryarchaeota\" Thermoplasmata Thermop...      2   \n",
       "MID1_17   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2   \n",
       "MID1_18   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1   \n",
       "MID1_19   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1   \n",
       "MID1_20   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1   \n",
       "MID1_21   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2   \n",
       "MID1_22   Bacteria \"Actinobacteria\" Actinobacteria Actin...     53   \n",
       "MID1_23   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1   \n",
       "MID1_24   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2   \n",
       "MID1_25   Bacteria \"Actinobacteria\" Actinobacteria Actin...      3   \n",
       "MID1_26   Bacteria \"Actinobacteria\" Actinobacteria Actin...      1   \n",
       "MID1_27   Bacteria \"Actinobacteria\" Actinobacteria Actin...     99   \n",
       "MID1_28   Bacteria \"Actinobacteria\" Actinobacteria Actin...     17   \n",
       "MID1_29   Bacteria \"Actinobacteria\" Actinobacteria Actin...      2   \n",
       "...                                                     ...    ...   \n",
       "MID9_251  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     10   \n",
       "MID9_252  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_253  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_254  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_255  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     12   \n",
       "MID9_256  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      4   \n",
       "MID9_257  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    805   \n",
       "MID9_258  Bacteria \"Proteobacteria\" Gammaproteobacteria ...    862   \n",
       "MID9_259  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     12   \n",
       "MID9_260  Bacteria \"Proteobacteria\" Gammaproteobacteria ...   3534   \n",
       "MID9_261  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      4   \n",
       "MID9_262  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_263  Bacteria \"Proteobacteria\" Gammaproteobacteria ...     25   \n",
       "MID9_264  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2   \n",
       "MID9_265  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      3   \n",
       "MID9_266  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2   \n",
       "MID9_267  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_268  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      3   \n",
       "MID9_269  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      2   \n",
       "MID9_270  Bacteria \"Proteobacteria\" Gammaproteobacteria ...      1   \n",
       "MID9_271  Bacteria \"Synergistetes\" Synergistia Synergist...      3   \n",
       "MID9_272  Bacteria \"Tenericutes\" Mollicutes Anaeroplasma...      1   \n",
       "MID9_273  Bacteria \"Tenericutes\" Mollicutes Haloplasmata...     11   \n",
       "MID9_274  Bacteria \"Thermodesulfobacteria\" Thermodesulfo...      1   \n",
       "MID9_275  Bacteria \"Thermodesulfobacteria\" Thermodesulfo...      1   \n",
       "MID9_276  Bacteria \"Thermotogae\" Thermotogae Thermotogal...     43   \n",
       "MID9_277  Bacteria \"Thermotogae\" Thermotogae Thermotogal...     16   \n",
       "MID9_278  Bacteria \"Verrucomicrobia\" Opitutae Opitutales...      1   \n",
       "MID9_279  Bacteria Cyanobacteria Cyanobacteria  Chloropl...      1   \n",
       "MID9_280  Bacteria Cyanobacteria Cyanobacteria  Chloropl...     10   \n",
       "\n",
       "                       Group   Sample  \n",
       "BARCODE                                \n",
       "MID1_0    EXTRACTION CONTROL  unknown  \n",
       "MID1_1    EXTRACTION CONTROL  unknown  \n",
       "MID1_2    EXTRACTION CONTROL  unknown  \n",
       "MID1_3    EXTRACTION CONTROL  unknown  \n",
       "MID1_4    EXTRACTION CONTROL  unknown  \n",
       "MID1_5    EXTRACTION CONTROL  unknown  \n",
       "MID1_6    EXTRACTION CONTROL  unknown  \n",
       "MID1_7    EXTRACTION CONTROL  unknown  \n",
       "MID1_8    EXTRACTION CONTROL  unknown  \n",
       "MID1_9    EXTRACTION CONTROL  unknown  \n",
       "MID1_10   EXTRACTION CONTROL  unknown  \n",
       "MID1_11   EXTRACTION CONTROL  unknown  \n",
       "MID1_12   EXTRACTION CONTROL  unknown  \n",
       "MID1_13   EXTRACTION CONTROL  unknown  \n",
       "MID1_14   EXTRACTION CONTROL  unknown  \n",
       "MID1_15   EXTRACTION CONTROL  unknown  \n",
       "MID1_16   EXTRACTION CONTROL  unknown  \n",
       "MID1_17   EXTRACTION CONTROL  unknown  \n",
       "MID1_18   EXTRACTION CONTROL  unknown  \n",
       "MID1_19   EXTRACTION CONTROL  unknown  \n",
       "MID1_20   EXTRACTION CONTROL  unknown  \n",
       "MID1_21   EXTRACTION CONTROL  unknown  \n",
       "MID1_22   EXTRACTION CONTROL  unknown  \n",
       "MID1_23   EXTRACTION CONTROL  unknown  \n",
       "MID1_24   EXTRACTION CONTROL  unknown  \n",
       "MID1_25   EXTRACTION CONTROL  unknown  \n",
       "MID1_26   EXTRACTION CONTROL  unknown  \n",
       "MID1_27   EXTRACTION CONTROL  unknown  \n",
       "MID1_28   EXTRACTION CONTROL  unknown  \n",
       "MID1_29   EXTRACTION CONTROL  unknown  \n",
       "...                      ...      ...  \n",
       "MID9_251           Control 2    stool  \n",
       "MID9_252           Control 2    stool  \n",
       "MID9_253           Control 2    stool  \n",
       "MID9_254           Control 2    stool  \n",
       "MID9_255           Control 2    stool  \n",
       "MID9_256           Control 2    stool  \n",
       "MID9_257           Control 2    stool  \n",
       "MID9_258           Control 2    stool  \n",
       "MID9_259           Control 2    stool  \n",
       "MID9_260           Control 2    stool  \n",
       "MID9_261           Control 2    stool  \n",
       "MID9_262           Control 2    stool  \n",
       "MID9_263           Control 2    stool  \n",
       "MID9_264           Control 2    stool  \n",
       "MID9_265           Control 2    stool  \n",
       "MID9_266           Control 2    stool  \n",
       "MID9_267           Control 2    stool  \n",
       "MID9_268           Control 2    stool  \n",
       "MID9_269           Control 2    stool  \n",
       "MID9_270           Control 2    stool  \n",
       "MID9_271           Control 2    stool  \n",
       "MID9_272           Control 2    stool  \n",
       "MID9_273           Control 2    stool  \n",
       "MID9_274           Control 2    stool  \n",
       "MID9_275           Control 2    stool  \n",
       "MID9_276           Control 2    stool  \n",
       "MID9_277           Control 2    stool  \n",
       "MID9_278           Control 2    stool  \n",
       "MID9_279           Control 2    stool  \n",
       "MID9_280           Control 2    stool  \n",
       "\n",
       "[2396 rows x 4 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metaData = pd.read_excel('Data/microbiome/metadata.xls',sheetname=0)\n",
    "list_of_tables_RNA = [pd.read_excel('Data/microbiome/MID' + str(n)+'.xls',header=None,columns=['a','b']) for n in range(1,10)];\n",
    "for i in range(0,9):\n",
    "    for j in range(0,3):\n",
    "        list_of_tables_RNA[i][metaData.columns[j]]=metaData[metaData.columns[j]][i]\n",
    "RNA = pd.concat(list_of_tables_RNA)\n",
    "ID = RNA.BARCODE + '_'+RNA.index.astype(str)\n",
    "RNA.index = ID\n",
    "RNA=RNA.drop(\"BARCODE\",axis=1)\n",
    "RNA.columns=['Name','Count','Group','Sample']\n",
    "RNA = RNA.replace(np.nan, 'unknown')\n",
    "RNA\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'/titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Write your answer here"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
