{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Table of Contents\n",
    " <p><div class=\"lev1\"><a href=\"#Task-1.-Compiling-Ebola-Data\"><span class=\"toc-item-num\">Task 1.&nbsp;&nbsp;</span>Compiling Ebola Data</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-2.-RNA-Sequences\"><span class=\"toc-item-num\">Task 2.&nbsp;&nbsp;</span>RNA Sequences</a></div>\n",
    " <div class=\"lev1\"><a href=\"#Task-3.-Class-War-in-Titanic\"><span class=\"toc-item-num\">Task 3.&nbsp;&nbsp;</span>Class War in Titanic</a></div></p>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from dateutil.parser import parse"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA_FOLDER = './data/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 1. Compiling Ebola Data\n",
    "\n",
    "The `DATA_FOLDER/ebola` folder contains summarized reports of Ebola cases from three countries (Guinea, Liberia and Sierra Leone) during the recent outbreak of the disease in West Africa. For each country, there are daily reports that contain various information about the outbreak in several cities in each country.\n",
    "\n",
    "Use pandas to import these data files into a single `Dataframe`.\n",
    "Using this `DataFrame`, calculate for *each country*, the *daily average* per year of *new cases* and *deaths*.\n",
    "Make sure you handle all the different expressions for *new cases* and *deaths* that are used in the reports."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- By studying the data we decided to only choose the confirmed cases and deaths. The reason herefor is that there is no way to know, if a patient who is previously registered as a suspected case, gets counted again when his illness is confirmed to be ebola. Analogous for the deaths, maybe one day a death gets registered as potentially ebola and days later after more thorough medical analysis it is confirmed to have been ebola, so another death gets counted in the confirmed cases. We did not want this to occur.\n",
    "\n",
    "\n",
    "\n",
    "1. Create new Data Frame with wanted attributes\n",
    "2. Provide functions that reads all csvs from a given directory\n",
    "3. Append wanted attributes to new data frame (only confirmed)\n",
    "4. Group data by Month and country and calculate the mean on the other fields. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2014-08-12-v77.csv 264 0\n",
      "2014-08-13-v78.csv 273 264\n",
      "2014-08-14-v79.csv 280 273\n",
      "2014-08-15-v80.csv 287 280\n",
      "2014-08-16-v81.csv 297 287\n",
      "2014-08-17-v82.csv 305 297\n",
      "2014-08-18-v83.csv 312 305\n",
      "2014-08-19-v84.csv 320 312\n",
      "2014-08-20-v85.csv 322 320\n",
      "2014-08-21-v86.csv 329 322\n",
      "2014-08-22-v87.csv 333 329\n",
      "2014-08-23-v88.csv 336 333\n",
      "2014-08-24-v89.csv 341 336\n",
      "2014-08-25-v90.csv 355 341\n",
      "2014-08-26-v91.csv 0 355\n",
      "2014-08-27-v92.csv 372 355\n",
      "2014-08-28-v93.csv 377 372\n",
      "2014-08-29-v94.csv 0 377\n",
      "2014-08-30-v95.csv 387 377\n",
      "2014-08-31-v96.csv 387 387\n",
      "2014-09-01-v97.csv 396 387\n",
      "2014-09-02-v98.csv 399 396\n",
      "2014-09-03-v99.csv 404 399\n",
      "2014-09-05-v101.csv 0 404\n",
      "2014-09-06-v102.csv 426 404\n",
      "2014-09-07-v103.csv 428 426\n",
      "2014-09-08-v104.csv 433 428\n",
      "2014-09-09-v105.csv 436 433\n",
      "2014-09-10-v106.csv 445 436\n",
      "2014-09-11-v107.csv 450 445\n",
      "2014-09-12-v108.csv 0 450\n",
      "2014-09-13-v109.csv 463 450\n",
      "2014-09-14-v110.csv 468 463\n",
      "2014-09-15-v111.csv 474 468\n",
      "2014-09-16-v112.csv 483 474\n",
      "2014-09-17-v113.csv 489 483\n",
      "2014-09-18-v114.csv 495 489\n",
      "2014-09-19-v115.csv 497 495\n",
      "2014-09-20-v116.csv 501 497\n",
      "2014-09-21-v117.csv 502 501\n",
      "2014-09-22-v118.csv 506 502\n",
      "2014-09-23-v119.csv 509 506\n",
      "2014-09-24-v120.csv 509 509\n",
      "2014-09-25-v121.csv 513 509\n",
      "2014-09-26-v122.csv 513 513\n",
      "2014-09-27-v123.csv 518 513\n",
      "2014-09-28-v124.csv 522 518\n",
      "2014-09-29-v125.csv 527 522\n",
      "2014-09-30-v126.csv 550 527\n",
      "2014-10-01-v127.csv 532 550\n",
      "2014-10-03-v129.csv 538 532\n",
      "2014-10-04-v130.csv 678 538\n",
      "2014-10-05-v131.csv 699 678\n",
      "2014-10-06-v132.csv 703 699\n",
      "2014-10-07-v133.csv 708 703\n",
      "2014-10-08-v134.csv 713 708\n",
      "2014-10-10-v136.csv 904 713\n",
      "2014-10-11-v137.csv 921 904\n",
      "2014-10-12-v138.csv 926 921\n",
      "2014-10-13-v139.csv 931 926\n",
      "2014-10-14-v140.csv 932 931\n",
      "2014-10-16-v141.csv 943 932\n",
      "2014-10-16-v142.csv 947 943\n",
      "2014-10-17-v143.csv 954 947\n",
      "2014-10-18-v144.csv 973 954\n",
      "2014-10-19-v145.csv 986 973\n",
      "2014-10-20-v146.csv 997 986\n",
      "2014-10-21-v147.csv 1001 997\n",
      "2014-10-22-v148.csv 1008 1001\n",
      "2014-10-23-v149.csv 1012 1008\n",
      "2014-10-24-v150.csv 1026 1012\n",
      "2014-10-25-v151.csv 1037 1026\n",
      "2014-10-26-v152.csv 1044 1037\n",
      "2014-10-27-v153.csv 1049 1044\n",
      "2014-10-28-v154.csv 1057 1049\n",
      "2014-10-30-v156.csv 1064 1057\n",
      "2014-10-31-v157.csv 1070 1064\n",
      "2014-11-01-v158.csv 1077 1070\n",
      "2014-11-02-v159.csv 1085 1077\n",
      "2014-11-06-v163.csv 1114 1085\n",
      "2014-11-07-v164.csv 1126 1114\n",
      "2014-11-08-v165.csv 1133 1126\n",
      "2014-11-10-v167.csv 1149 1133\n",
      "2014-11-12-v169.csv 1169 1149\n",
      "2014-11-13.csv 1180 1169\n",
      "2014-11-14.csv 0 1180\n",
      "2014-11-15.csv 1206 1180\n",
      "2014-11-16.csv 1223 1206\n",
      "2014-11-17.csv 1223 1223\n",
      "2014-11-18-v175.csv 1240 1223\n",
      "2014-11-19-v176.csv 0 1240\n",
      "2014-11-20-v177.csv 1282 1240\n",
      "2014-11-21-v178.csv 1303 1282\n",
      "2014-11-22-v179.csv 1333 1303\n",
      "2014-11-23-v180.csv 1364 1333\n",
      "2014-11-24-v181.csv 1397 1364\n",
      "2014-11-28.csv 1481 1397\n",
      "2014-11-29.csv 1496 1481\n",
      "2014-12-01.csv 1549 1496\n",
      "2014-12-04.csv 1648 1549\n",
      "2014-12-05.csv 1669 1648\n",
      "2014-12-06.csv 1708 1669\n",
      "2014-12-13.csv 0 1708\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>New</th>\n",
       "      <th>Deaths</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Month</th>\n",
       "      <th>Country</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <th>Liberia</th>\n",
       "      <td>2.142857</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <th>Liberia</th>\n",
       "      <td>1.818182</td>\n",
       "      <td>4.272727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">8</th>\n",
       "      <th>Guinea</th>\n",
       "      <td>12.400000</td>\n",
       "      <td>3.400000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liberia</th>\n",
       "      <td>5.444444</td>\n",
       "      <td>23.222222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sierra Leone</th>\n",
       "      <td>18.550000</td>\n",
       "      <td>19.350000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">9</th>\n",
       "      <th>Guinea</th>\n",
       "      <td>13.000000</td>\n",
       "      <td>3.562500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liberia</th>\n",
       "      <td>6.166667</td>\n",
       "      <td>36.041667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sierra Leone</th>\n",
       "      <td>34.413793</td>\n",
       "      <td>5.620690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"3\" valign=\"top\">10</th>\n",
       "      <th>Guinea</th>\n",
       "      <td>6.000000</td>\n",
       "      <td>15.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Liberia</th>\n",
       "      <td>1.360000</td>\n",
       "      <td>28.040000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sierra Leone</th>\n",
       "      <td>56.714286</td>\n",
       "      <td>18.571429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">11</th>\n",
       "      <th>Liberia</th>\n",
       "      <td>2.600000</td>\n",
       "      <td>13.466667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sierra Leone</th>\n",
       "      <td>59.904762</td>\n",
       "      <td>20.285714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <th>Sierra Leone</th>\n",
       "      <td>32.600000</td>\n",
       "      <td>42.400000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          New     Deaths\n",
       "Month Country                           \n",
       "6     Liberia        2.142857   2.000000\n",
       "7     Liberia        1.818182   4.272727\n",
       "8     Guinea        12.400000   3.400000\n",
       "      Liberia        5.444444  23.222222\n",
       "      Sierra Leone  18.550000  19.350000\n",
       "9     Guinea        13.000000   3.562500\n",
       "      Liberia        6.166667  36.041667\n",
       "      Sierra Leone  34.413793   5.620690\n",
       "10    Guinea         6.000000  15.000000\n",
       "      Liberia        1.360000  28.040000\n",
       "      Sierra Leone  56.714286  18.571429\n",
       "11    Liberia        2.600000  13.466667\n",
       "      Sierra Leone  59.904762  20.285714\n",
       "12    Sierra Leone  32.600000  42.400000"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ebola = pd.DataFrame(columns=['Month', 'Country', 'New', 'Deaths'])\n",
    "\n",
    "def ebola_dirs(dirname, country, desc_col, totals_col, new_id, deaths_id):\n",
    "    global ebola\n",
    "    \n",
    "    path = DATA_FOLDER + 'ebola/' + dirname\n",
    "    old_sierra = 0\n",
    "    for csv in sorted(os.listdir(path)):\n",
    "        frame = pd.read_csv(path + '/' + csv).fillna(0)\n",
    "        frame = frame.replace(to_replace='.*%$', value='0', regex=True)\n",
    "        frame = frame.replace(to_replace='.*\\..*$', value='0', regex=True)\n",
    "        frame = frame.replace(to_replace='.*\\,.*', value='0', regex=True)\n",
    "        \n",
    "        if country == 'Liberia' and '2014-12' in csv:\n",
    "            continue\n",
    "        \n",
    "        deaths_id_mod = deaths_id\n",
    "        if country == 'Guinea' and csv == '2014-08-04.csv':\n",
    "            deaths_id_mod += ' today'\n",
    "        \n",
    "        month = parse(frame[frame.columns[0]][0]).month\n",
    "        new = int(frame[frame[desc_col] == new_id][totals_col])\n",
    "        deaths = 0\n",
    "        if country == 'Sierra Leone':\n",
    "            deaths1 = int(frame[frame[desc_col] == deaths_id_mod][totals_col])\n",
    "            print(csv, deaths1, old_sierra)\n",
    "            if deaths1 != 0:\n",
    "                deaths = deaths1 - old_sierra\n",
    "                old_sierra = deaths1\n",
    "            else:\n",
    "                deaths = 0\n",
    "        else:\n",
    "            deaths = int(frame[frame[desc_col] == deaths_id_mod][totals_col])\n",
    "        \n",
    "        row = pd.DataFrame({'Month': month, 'Country': country, 'New': new, 'Deaths': deaths}, index=[0])\n",
    "        ebola = ebola.append(row, ignore_index=True)\n",
    "\n",
    "ebola_dirs('guinea_data/', 'Guinea', 'Description', 'Totals', 'New cases of confirmed', 'New deaths registered')\n",
    "ebola_dirs('liberia_data/', 'Liberia', 'Variable', 'National', 'New case/s (confirmed)', 'Newly reported deaths')\n",
    "ebola_dirs('sl_data/', 'Sierra Leone', 'variable', 'National', 'new_confirmed', 'death_confirmed')\n",
    "\n",
    "ebola[\"Deaths\"] = pd.to_numeric(ebola[\"Deaths\"])\n",
    "ebola[\"New\"] = pd.to_numeric(ebola[\"New\"])\n",
    "ebola.groupby(['Month', 'Country']).agg({'New': np.mean, 'Deaths': np.mean})\n",
    "#ebola"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 2. RNA Sequences\n",
    "\n",
    "In the `DATA_FOLDER/microbiome` subdirectory, there are 9 spreadsheets of microbiome data that was acquired from high-throughput RNA sequencing procedures, along with a 10<sup>th</sup> file that describes the content of each. \n",
    "\n",
    "Use pandas to import the first 9 spreadsheets into a single `DataFrame`.\n",
    "Then, add the metadata information from the 10<sup>th</sup> spreadsheet as columns in the combined `DataFrame`.\n",
    "Make sure that the final `DataFrame` has a unique index and all the `NaN` values have been replaced by the tag `unknown`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each excel file, we:\n",
    "    - Read the excel file into a pandas data frame\n",
    "    - Name the columns and append a new Barcode column to it, where every entry is the name of the current file\n",
    "    - Append this data frame to the rna data frame collecting the data of all 10 files\n",
    "    \n",
    "Then read the meta excel file into a data frame and do an inner join with the rna data frame on the \"BARCODE\" column. We replaced all the NaN values with \"unknown\" and create a unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Task 2\n",
    "\n",
    "rna = pd.DataFrame()\n",
    "\n",
    "for i in range(1, 10):\n",
    "    file = 'MID' + str(i)\n",
    "    frame = pd.read_excel(DATA_FOLDER + '/microbiome/' + file + '.xls')\n",
    "    frame.columns = ['TYPE', 'NUMBER']\n",
    "    frame['BARCODE'] = file\n",
    "    frame.index = list(map(lambda n: file + '_' + str(n), np.arange(1, len(frame)+1)))\n",
    "    rna = pd.concat([rna, frame])\n",
    "    \n",
    "meta = pd.read_excel(DATA_FOLDER + '/microbiome/metadata.xls', index_col=0)\n",
    "\n",
    "rna.join(meta, on=\"BARCODE\").drop('BARCODE', axis=1).fillna('unknown')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Task 3. Class War in Titanic\n",
    "\n",
    "Use pandas to import the data file `Data/titanic.xls`. It contains data on all the passengers that travelled on the Titanic."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.core.display import HTML\n",
    "HTML(filename=DATA_FOLDER+'titanic.html')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For each of the following questions state clearly your assumptions and discuss your findings:\n",
    "1. Describe the *type* and the *value range* of each attribute. Indicate and transform the attributes that can be `Categorical`. \n",
    "2. Plot histograms for the *travel class*, *embarkation port*, *sex* and *age* attributes. For the latter one, use *discrete decade intervals*. \n",
    "3. Calculate the proportion of passengers by *cabin floor*. Present your results in a *pie chart*.\n",
    "4. For each *travel class*, calculate the proportion of the passengers that survived. Present your results in *pie charts*.\n",
    "5. Calculate the proportion of the passengers that survived by *travel class* and *sex*. Present your results in *a single histogram*.\n",
    "6. Create 2 equally populated *age categories* and calculate survival proportions by *age category*, *travel class* and *sex*. Present your results in a `DataFrame` with unique index."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "titanic = pd.read_excel(DATA_FOLDER + '/titanic.xls')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3.1\n",
    "\n",
    "| Attribute     | Type     | Range         | Categorical     |\n",
    "|---------------|----------|---------------|-----------------|\n",
    "|pclass         |int64     |[1, 3]         |Yes              |\n",
    "|survived       |int64     |[0, 1]         |Yes              |\n",
    "|name           |object    |-              |No               |\n",
    "|sex            |object    |{male, female} |Yes              |\n",
    "|age            |float64   |[0.1667, 80.0] |No               |\n",
    "|sibsp          |int64     |[0, 8]         |No               |\n",
    "|parch          |int64     |[0, 9]         |No               |\n",
    "|ticket         |object    |-              |Yes              |\n",
    "|fare           |float64   |[0.0, 512.33]  |No               |\n",
    "|cabin          |object    |-              |Yes              |\n",
    "|embarked       |object    |{S, C, Q}      |Yes              |\n",
    "|boat           |object    |-              |Yes              |\n",
    "|body           |float64   |[1.0, 328.0]   |No               |\n",
    "|home.dest      |object    |-              |No               |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "titanic.pclass = titanic.pclass.astype('category')\n",
    "titanic.survived = titanic.survived.astype('category')\n",
    "titanic.ticket = titanic.ticket.astype('category')\n",
    "titanic.cabin = titanic.cabin.astype('category')\n",
    "titanic.sex = titanic.sex.astype('category')\n",
    "titanic.boat = titanic.boat.astype('category')\n",
    "titanic.embarked = titanic.embarked.astype('category')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Miscellaneous histograms\n",
    "\n",
    "plt.figure(figsize=(16, 4))\n",
    "\n",
    "# Travel class\n",
    "plt.subplot(141)\n",
    "tc = titanic.pclass.value_counts().plot(kind='bar')\n",
    "\n",
    "# Embarkation point\n",
    "titanic.embarked.replace(['S', 'C', 'Q'], ['Southampton', 'Cherbourg', 'Queenstown'], inplace=True)\n",
    "plt.subplot(142)\n",
    "em = titanic.embarked.value_counts().plot(kind='bar')\n",
    "\n",
    "# Sex \n",
    "plt.subplot(143)\n",
    "sx = titanic.sex.value_counts().plot(kind='bar')\n",
    "\n",
    "# Age\n",
    "plt.subplot(144)\n",
    "ae = titanic.age.hist(bins=np.arange(0, 100, 10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 3. Cabin floor allocation\n",
    "\n",
    "titanic.cabin.dropna().map(lambda cf: str(cf)[0]).value_counts().plot.pie(autopct='%.1f%%', pctdistance=1.1, labeldistance=1.3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#titanic[['pclass', 'survived']].stack().astype(int).unstack()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# 4. Survivors per cabin floor\n",
    "\n",
    "sv = titanic[['pclass', 'survived']].stack().astype(int).unstack().groupby(['pclass']).mean()\n",
    "\n",
    "plt.figure(figsize=(16, 4))\n",
    "\n",
    "plt.subplot(131)\n",
    "plt.pie([sv.loc[1], 1 - sv.loc[1]], labels=['a', 'b'], autopct='%.1f%%', pctdistance=1.1, labeldistance=1.3)\n",
    "\n",
    "plt.subplot(132)\n",
    "plt.pie([sv.loc[2], 1 - sv.loc[2]])\n",
    "\n",
    "plt.subplot(133)\n",
    "plt.pie([sv.loc[3], 1 - sv.loc[3]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sv = titanic[['survived', 'pclass', 'sex']].groupby(['pclass', 'sex']).sum()\n",
    "st = titanic[['pclass', 'sex']].groupby(['pclass', 'sex']).size()\n",
    "sv['died'] = st-sv.survived\n",
    "sv.plot.bar(stacked=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>survived</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pclass</th>\n",
       "      <th>sex</th>\n",
       "      <th>cat</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">1</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">female</th>\n",
       "      <th>old</th>\n",
       "      <td>0.967391</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.951220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">male</th>\n",
       "      <th>old</th>\n",
       "      <td>0.302521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.531250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">2</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">female</th>\n",
       "      <th>old</th>\n",
       "      <td>0.872340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.910714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">male</th>\n",
       "      <th>old</th>\n",
       "      <td>0.083333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.216216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"4\" valign=\"top\">3</th>\n",
       "      <th rowspan=\"2\" valign=\"top\">female</th>\n",
       "      <th>old</th>\n",
       "      <td>0.372093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.513761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th rowspan=\"2\" valign=\"top\">male</th>\n",
       "      <th>old</th>\n",
       "      <td>0.144000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>young</th>\n",
       "      <td>0.183036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     survived\n",
       "pclass sex    cat            \n",
       "1      female old    0.967391\n",
       "              young  0.951220\n",
       "       male   old    0.302521\n",
       "              young  0.531250\n",
       "2      female old    0.872340\n",
       "              young  0.910714\n",
       "       male   old    0.083333\n",
       "              young  0.216216\n",
       "3      female old    0.372093\n",
       "              young  0.513761\n",
       "       male   old    0.144000\n",
       "              young  0.183036"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cat = titanic.sort_values('age')[['age', 'survived', 'sex', 'pclass']].dropna()\n",
    "cat['cat'] = pd.qcut(cat['age'].values, 2, labels=['young', 'old'])\n",
    "cat = cat[['cat', 'survived', 'pclass', 'sex']]\n",
    "cat[\"survived\"] = pd.to_numeric(cat[\"survived\"])\n",
    "cat.groupby(['pclass', 'sex', 'cat']).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda root]",
   "language": "python",
   "name": "conda-root-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
